common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -140,6 +140,52 @@ ICacheRecord save(@NonNull AccountRecord accountRecord,
         return result;
     }
 
+    // TODO Add unit test
+    @NonNull
+    List<ICacheRecord> saveAndLoadAggregatedAccountData(
+            @NonNull AccountRecord accountRecord,
+            @NonNull IdTokenRecord idTokenRecord,
+            @NonNull AccessTokenRecord accessTokenRecord) throws ClientException {
+        return mergeCacheRecordWithCorollaryAccounts(
+                save(accountRecord, idTokenRecord, accessTokenRecord)
+        );
+    }
+
+    @NonNull
+    private List<ICacheRecord> mergeCacheRecordWithCorollaryAccounts(
+            @NonNull final ICacheRecord savedCacheRecord) {
+        final List<ICacheRecord> result = new ArrayList<>();
+        // Whatever ICacheRecord you provide will _always_ be the first element in the result List.
+        result.add(savedCacheRecord);
+
+        final List<AccountRecord> corollaryAccounts = new ArrayList<>(
+                getCorollaryAccounts(
+                        savedCacheRecord
+                                .getRefreshToken()
+                                .getClientId(),
+                        savedCacheRecord
+                                .getAccount() // This account wil be the 0th element in the result.
+                )
+        );
+
+        if (!corollaryAccounts.isEmpty()) {
+            // Remove the first element from the List since it is already contained in the result List
+            corollaryAccounts.remove(0);
```

Comment:
This entry was added in getCorollaryAccounts... and then removed here.  I think this may be due to the same method being used from 2 different paths.  Just confirming that this is necessary.

Reply:
Yep - this is an artifact of the code paths... I wanted the behavior of both `mergeCacheRecordWithCorollaryAccounts` and `getCorollaryAccounts` to be the same: to always return their input as the 0th element.

Since `mergeCacheRecordWithCorollaryAccounts` calls `getCorollaryAccounts` and they both return their input as the first element, I need to remove the first element of the result so I don't end up with the same record twice in the final result.

This could be factored out[ by only calling `result.add()` if `corollaryAccounts.isEmpty()`](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/iambmelt/tenant-profile/common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java#L159-L160) but I thought that would be less obvious.